libs/anglify/src/modules/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, ChangeDetectionStrategy, AfterViewInit, forwardRef } from '@angular/core';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { BehaviorSubject, combineLatest, debounceTime, fromEvent, map, ReplaySubject, share } from 'rxjs';
+import { createSettingsProvider } from '../../factories/settings.factory';
+import { DEFAULT_SELECT_SETTINGS, SELECT_SETTINGS } from '../select/select-settings.token';
+import { SelectComponent } from '../select/select.component';
+import { EntireSelectSettings } from '../select/select.interface';
+
+@UntilDestroy()
+@Component({
+  selector: 'anglify-autocomplete',
+  templateUrl: './autocomplete.component.html',
+  styleUrls: ['./autocomplete.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => AutocompleteComponent),
+      multi: true,
+    },
+    createSettingsProvider<EntireSelectSettings>('anglifySelectSettings', DEFAULT_SELECT_SETTINGS, SELECT_SETTINGS),
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AutocompleteComponent extends SelectComponent implements AfterViewInit {
+  private readonly inputValue$ = new BehaviorSubject<string>('');
+
+  public readonly filteredOptions$ = combineLatest([this.options$, this.inputValue$]).pipe(
+    untilDestroyed(this),
+    map(([items, value]) => (value ? items.filter(item => item.text.includes(value)) : items)),
+    share({
+      connector: () => new ReplaySubject(1),
+    })
+  );
+
+  public override ngAfterViewInit() {
+    super.ngAfterViewInit();
+
+    this.menu.isOpen$.pipe(untilDestroyed(this)).subscribe(isOpen => {
+      if (isOpen) {
+        this.inputValue$.next(this.input.elementRef.nativeElement.value);
+      }
+    });
+
+    fromEvent<InputEvent & { target: HTMLInputElement }>(this.input.elementRef.nativeElement, 'input')
```

Comment:
Consider applying CQRS by splitting the streams into commands and queries. 

You can use the defer operator if you don't have access to the menu reference and the input reference.

The input value stream is a derived state, so you can remove the BehaviorSubject, which creates an internal state on its own.